src/super_gradients/training/pre_launch_callbacks/pre_launch_callbacks.py

```diff
@@ -250,7 +254,8 @@ def __call__(self, cfg: Union[dict, DictConfig]) -> Union[dict, DictConfig]:
             cfg.num_gpus = 1
 
         # no augmentations
-        if "transforms" in cfg.dataset_params.val_dataset_params:
+        if self.disable_augmentations and "transforms" in cfg.dataset_params.val_dataset_params:
+            logger.warning("Augmentations will be disabled for QAT run.")
             cfg.dataset_params.train_dataset_params.transforms = cfg.dataset_params.val_dataset_params.transforms
```

Comment:
I think, collate_fn-based augmentations are a little out of scope of the rules-of-thumb that this callback implements. 
Do I just copy it from val_dataloader as with transforms?

Reply:
> Do I just copy it from val_dataloader as with transforms?
Yes, it should work. 

But there are additional edge cases that suggested change would not cover:
In detection on COCO dataset there is `with_crowd: True|False` property of Dataset config (Not transform). Depending on the flag value a different collate function should be used (DetectionCollate vs DetectionCollateWithCrowd). 

```
val_dataset_params:
  data_dir: /data/coco # root path to coco data
  subdir: images/val2017 # sub directory path of data_dir containing the train data.
  json_file: instances_val2017.json # path to coco train json file, data_dir/annotations/train_json_file.
  input_dim: [640, 640]
  cache_dir:
  cache: False
  transforms:
  - DetectionPaddedRescale:
      input_dim: ${dataset_params.val_dataset_params.input_dim}
  - DetectionTargetsFormatTransform:
      max_targets: 50
      input_dim: ${dataset_params.val_dataset_params.input_dim}
      output_format: LABEL_CXCYWH
  tight_box_rotation: False
  class_inclusion_list:
  max_num_samples:
  with_crowd: True                                                                  <----- THIS

val_dataloader_params:
  batch_size: 64
  num_workers: 8
  drop_last: False
  pin_memory: True
  collate_fn: # collate function for valset
    _target_: super_gradients.training.utils.detection_utils.CrowdDetectionCollateFN <----- THIS
```

So if you update the transforms from val -> train and update collate function as well, `with_crowd` would still be `False` for train dataset and would crash. 

I honestly don't see a 'nice' way to tackle this other than having an explicit config for calibration. Btw, why can't we use validation set of calibration?

Second question: maybe we add new sections to dataset config "calibration_dataset/calibration_loader"? This would simplify the logic and make black magic stuff (disabling augs) redundant.

Reply:
We do have a calibration loader now in QATTrainer
The intended purpose of this callback is to serve as a starting point for someone who just wants to add one line to his existing recipe and be able to get a decent QAT to start with. 
In my opinion, we can tolerate (some) of the augmentations introduced by collate_fn.

Also, I've only recently found out that calibrating without or with smaller number of augs and training with all of them is beneficial, that's why this PR.